misc/python/materialize/mzcompose/__init__.py

```diff
@@ -575,6 +608,26 @@ def run(
             capture=capture,
         )
 
+    def exec(
+        self, service: "Service", *args: str, detach: bool = False
+    ) -> subprocess.CompletedProcess:
+        """Execute a one-off command in a service's running container
+        Delegates to `docker exec` as `docker-compose exec` appears to do nothing
```

Comment:
How weird! This diff to use `docker-compose exec` works perfectly for me?

```diff
diff --git a/misc/python/materialize/mzcompose/__init__.py b/misc/python/materialize/mzcompose/__init__.py
index a84e297be..4102a318e 100644
--- a/misc/python/materialize/mzcompose/__init__.py
+++ b/misc/python/materialize/mzcompose/__init__.py
@@ -372,39 +372,6 @@ class Composition:
                 print(e.stdout)
             raise UIError(f"running docker-compose failed (exit status {e.returncode})")
 
-    def invoke_docker(
-        self, *args: str, capture: bool = False
-    ) -> subprocess.CompletedProcess:
-        """Invoke `docker` directly
-
-        Args:
-            args: The arguments to pass to `docker`.
-            capture: Whether to capture the child's stdout stream.
-        """
-
-        if not self.silent:
-            print(f"$ docker {' '.join(args)}", file=sys.stderr)
-
-        stdout = None
-        if capture:
-            stdout = subprocess.PIPE
-
-        try:
-            return subprocess.run(
-                [
-                    "docker",
-                    *args,
-                ],
-                close_fds=False,
-                check=True,
-                stdout=stdout,
-                text=True,
-            )
-        except subprocess.CalledProcessError as e:
-            if e.stdout:
-                print(e.stdout)
-            raise UIError(f"running docker failed (exit status {e.returncode})")
-
     def port(self, service: str, private_port: Union[int, str]) -> int:
         """Get the public port for a service's private port.
 
@@ -612,19 +579,19 @@ class Composition:
         self, service: "Service", *args: str, detach: bool = False
     ) -> subprocess.CompletedProcess:
         """Execute a one-off command in a service's running container
-        Delegates to `docker exec` as `docker-compose exec` appears to do nothing
+
+        Delegates to `docker-compose exec`.
+
         Args:
             service: The service whose container will be used
             command: The command to run
             args: Arguments to pass to the command
         """
 
-        container = "_".join([self.name, service.name, "1"])
-
-        return self.invoke_docker(
+        return self.invoke(
             "exec",
             *(["--detach"] if detach else []),
-            container,
+            service.name,
             *args,
         )
 
```

Reply:
I know what happened -- I was running `docker-compose exec` manually without a mzcompose.yml file present -- in this case, the command does nothing and returns nothing with no error message :-( .